src/mme/mme-gn-build.c

```diff
@@ -212,7 +212,7 @@ ogs_pkbuf_t *mme_gn_build_sgsn_context_request(
     req->temporary_logical_link_identifier.presence = 0;
 
     req->packet_tmsi.presence = 1;
-    req->packet_tmsi.u32 = be32toh(ptmsi);
+    req->packet_tmsi.u32 = ptmsi;
```

Comment:
But the PTMSI is generated just above in line 204. So the generated PTMSI in there is actually generated  in host byte order?

Reply:
Yes, I believe so because the tlv functions that decode the TAU already convert the byte order.
`tlv_add_leaf()`/`tlv_parse_leaf()` in ogs-tlv-msg.c are what takes care to convert byte order during encode/decode respectively for elements of type `OGS_T*V_[U]INT16/24/32`.

At least that's how I understood it

Reply:
Ah I see, guti_to_rai_ptmsi() is probably generating the ptmsi in host byte order since it's not actually doing it byte per byte, so indeed the conversion is not needed here because it's done automatically by tlv functions.